### ISSUE TITLE
label-content-name-mismatch should not treat SVG titles as visible text

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4065]
This should not fail axe-core's `label-content-name-mismatch` rule:

```html
<a href="#" aria-label="Hello world">
  <svg width="14" height="14" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
    <title>Circle</title>
    <circle cx="50" cy="50" r="50" />
  </svg>
  Hello World
</a>
```

Axe considers the SVG title here to be part of the visible text, when it clearly isn't. This is because it calls `subtreeText`. I'm not sure whether the problem is that it calls `subtreeText` instead of something different, or that maybe `subtreeText` just shouldn't return SVG titles here.

**Comment by straker** [comment]
I'm thinking we should switch to use `visibleVirtual` to get the visible text content of the element as that only considers visible text nodes and does not process the accessible name of non-visible text nodes (like `aria-label` and svg `title`). It'll probably need to be modified though to ignore ligature fonts like `subtreeText` does and also handle adding whitespace between nodes like `<br/>` elements.

**Comment by chutchins25** [comment]
Fixed by #5096, which switched the check's visible-text calculation from `subtreeText` to `visibleVirtual` — exactly the approach @straker proposed above. `visibleVirtual` only considers visible text nodes and doesn't pull in accessible-name sources like an SVG `<title>`, so the title text is no longer treated as visible content.

Verified with a before/after build on the exact repro from this issue:
- pre-#5096 (`subtreeText`): fails (violation)
- post-#5096 (`visibleVirtual`): passes

Regression coverage shipped in the same PR — `#pass9` (this repro) and `#fail7` in the `label-content-name-mismatch` integration fixtures. Confirmed on axe-core 4.13.0. Closing as resolved.
